Problem_Set_5/ps5.py

```diff
@@ -0,0 +1,382 @@
+    
+    def evaluate(self, story):
+        return self.trigger1.evaluate(story) or self.trigger2.evaluate(story)
+
+#======================
+# Filtering
+#======================
+
+# Problem 10
+def filter_stories(stories, triggerlist):
+    """
+    Takes in a list of NewsStory instances.
+
+    Returns: a list of only the stories for which a trigger in triggerlist fires.
+    """
+    triggered_stories = []
+    for trigger in triggerlist:
+        for story in stories:
+            if trigger.evaluate(story) and story not in triggered_stories:
+                    triggered_stories.append(story)
+    return triggered_stories 
+
+
+
+
+#======================
+# User-Specified Triggers
+#======================
+# Problem 11
+def read_trigger_config(filename):
+    """
+    filename: the name of a trigger configuration file
+
+    Returns: a list of trigger objects specified by the trigger configuration
+        file.
+    """
+    # We give you the code to read in the file and eliminate blank lines and
+    # comments. You don't need to know how it works for now!
+    trigger_file = open(filename, 'r')
+    lines = []
+    for line in trigger_file:
+        line = line.rstrip()
+        if not (len(line) == 0 or line.startswith('//')):
+            lines.append(line)
+
+    # TODO: Problem 11
+    # line is the list of lines that you need to parse and for which you need
+    # to build triggers
+
+    ### Important note: didn't initially understand the question being asked so after googling the question
+    ### and seeing some guy's answer, I understood. But this is basically the same soln as his, cuz I would have done it same way.
+
+    trigger_dict = {}
+    trigger_list = []
+    for command in lines:
+        temp_list = [x for x in command.split(",")]
+        if temp_list[1] == "TITLE":
+            trigger_dict[temp_list[0]] = TitleTrigger(temp_list[2])
+        elif temp_list[1] == "DESCRIPTION":
+            trigger_dict[temp_list[0]] = DescriptionTrigger(temp_list[2])
+        elif temp_list[1] == "AFTER":
+            trigger_dict[temp_list[0]] = AfterTrigger(temp_list[2])
+        elif temp_list[1] == "BEFORE":
+            trigger_dict[temp_list[0]] = BeforeTrigger(temp_list[2])
+        elif temp_list[1] == "NOT":
+            trigger_dict[temp_list[0]] = NotTrigger(temp_list[2])
+        elif temp_list[1] == "AND":
+            trigger_dict[temp_list[0]] = AndTrigger(temp_list[2], temp_list[3])
+        elif temp_list[1] == "OR":
+            trigger_dict[temp_list[0]] = OrTrigger(temp_list[2], temp_list[3])
+        elif temp_list[0] == "ADD":
+            trigger_list.extend(temp_list[1:])
+    return trigger_list
+
+FILENAME = "Problem_Set_5/triggers.txt"
+
+SLEEPTIME = 120 #seconds -- how often we poll
+
+def main_thread(master):
+    # A sample trigger list - you might need to change the phrases to correspond
+    # to what is currently in the news
+    try:
+        t1 = TitleTrigger("COVID-19")
```

Comment:
Here, I'm supposed to comment out lines 317 to 321 and instead call my function read_trigger_config() to generate the triggerlist as I have in line 325. However when I run it with the function I get an error of "object has no attribute 'description'". So if you understand why this is happening and how i could fix it that would be great. But apart from this I think everything else is correct